File: ios/PaymentInfoUtility.swift
//
//  PaymentInfo.swift
//  Ecommpay
//
//  Created by Станислав Полещук on 22.07.2022.
//  Copyright © 2022 Facebook. All rights reserved.
//
import Foundation
import ecommpaySDK

public class PaymentInfoUtility {
  static func bind(info: NSDictionary) -> PaymentInfo {

    let projectId = info["projectId"] as? Int ?? 0
    let paymentId = info["paymentId"] as? String
    let paymentAmount = info["paymentAmount"] as? Int ?? 0
    let paymentCurrency = info["paymentCurrency"] as? String ?? "USD"
    let paymentDescription = info["paymentDescription"] as? String
    let customerId = info["customerId"] as? String
    let regionCode = info["regionCode"] as? String

    return PaymentInfo(
      projectID: projectId,
      paymentID: paymentId,
      paymentAmount: paymentAmount,
      paymentCurrency: paymentCurrency,
      paymentDescription: paymentDescription,
      customerID: customerId,
      regionCode: regionCode)
  }
}

File: android/src/main/java/com/reactnativeecommpay/utils/PaymentInfoUtility.java
package com.reactnativeecommpay.utils;

import com.ecommpay.sdk.ECMPPaymentInfo;
import com.facebook.react.bridge.ReadableMap;


public class PaymentInfoUtility {
  public static ECMPPaymentInfo bind(ReadableMap info) throws Exception {
    int projectId = info.hasKey("projectId") ? info.getInt("projectId") : 0;
    String paymentId = info.hasKey("paymentId") ? info.getString("paymentId") : null;
    long paymentAmount = info.hasKey("paymentAmount") ? TypeUtility.convertToLong(info.getDouble("paymentAmount")) : 0;
    String paymentCurrency = info.hasKey("paymentCurrency") ? info.getString("paymentCurrency") : null;
    String paymentDescription = info.hasKey("paymentDescription") ? info.getString("paymentDescription") : null;
    String customerId = info.hasKey("customerId") ? info.getString("customerId") : null;
    String regionCode = info.hasKey("regionCode") ? info.getString("regionCode") : null;


    if (TypeUtility.stringIsNotEmpty(paymentId) && TypeUtility.stringIsNotEmpty(paymentCurrency) &&
      TypeUtility.stringIsNotEmpty(paymentDescription) && TypeUtility.stringIsNotEmpty(customerId) &&
      TypeUtility.stringIsNotEmpty(regionCode)) {
      return new ECMPPaymentInfo(
        projectId,
        paymentId,
        paymentAmount,
        paymentCurrency,
        paymentDescription,
        customerId,
        regionCode);
    }

    if (TypeUtility.stringIsNotEmpty(paymentId) && TypeUtility.stringIsNotEmpty(paymentCurrency)) {
      return new ECMPPaymentInfo(
        projectId,
        paymentId,
        paymentAmount,
        paymentCurrency);
    }

    if (TypeUtility.stringIsNotEmpty(paymentCurrency)) {
      return new ECMPPaymentInfo(
        projectId,
        paymentAmount,
        paymentCurrency);
    }

    throw new Exception("Invalid payment info");
  }
}

